browser-extension/background.js

```diff
@@ -57,10 +57,10 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
       const threeWeeksAgo = getDateThreeWeeksAgo()
       const recent = await request_recommendations(`date_gte=${threeWeeksAgo}&language=${request.language}&limit=10`);
       const old = await request_recommendations(`date_lte=${threeWeeksAgo}&language=${request.language}&limit=50`);
-      const recent_sub = getRandomSubarray(recent, 3);
-      const old_sub = getRandomSubarray(old, 4 - recent_sub.length);
-      const videos = getRandomSubarray([...old_sub, ...recent_sub], 4);
-      return { data: videos };
+      const recent_sub = getRandomSubarray(recent, request.number - 1);
```

Comment:
I think we could discuss this algorithm together.

@aidanjungo @lfaucon @amatissart @FabienRoger 

If I understood correctly, when expanding the Tournesol recommendation panel, the code loads up to 7 recent videos, and 8 minus 7 older video(s).

So, when there is only one line of recommendations we have
- 3 recent videos (75 %)
- 1 older video (25 %)

When there is 3 lines of recommendations we expect to have:
- 10 recent videos (83%) 
- 2 older videos (17%)

The % of older videos keeps decreasing the more lines we loads. This leads me to several questions:

(1) Could this be a problem?

(2) Do we have enough pertinent contributions to trust the fact that the recommendations are still " quality recommendations " when we display three lines of videos with 83 % of recent videos?

Reply:
I agree. When I made this change I wasn't really thinking about the proportions of videos and I just did something "close enough". Thank you for catching that. I will revert to the old ratio and will properly implement something to make sure the ratio is respected both in the expanded and not-expanded form.

Reply:
Because we are showing more videos, let's also take them from a larger set of possible videos to show.

Currently we pick from 10 recent and 50 older videos. 

We are not yet confident enough in the quality of very recent videos so I prefer keeping it at 10 or 15, but we could increase the number of old videos to 200. 

Then it may be best to vary the proportions: first 4 stay the same but next 8 videos could be 75% old and 25% recent.

Happy to discuss more if needed, but we will iteratively improve this later too.

Reply:
> I agree. When I made this change I wasn't really thinking about the proportions of videos and I just did something "close enough". 

That's perfectly ok. These kinds of consequences are hard to spot when you are discovering the code. It's something we can keep in mind when we are developing: " ok now that the code is working, what are its consequences? " .

Reply:
I fixed that !